### ISSUE TITLE
FSM should be discarded on success/failure

### ISSUE THREAD

**Issue by ingesolvoll** [ingesolvoll/glimt#6]
We don't want dead interceptors to keep processing any event when FSM is finished. We could also clear db state, but would have to look into details there. Clearing state too early might harm UX.

**Comment by mainej** [comment]
> Clearing state too early might harm UX.

I agree that this is a big concern. I often want state to stick around (at least for a few seconds) after a request completes, so that I can flash a success (or failure) message or icon.

**Comment by mainej** [comment]
If you implement this, do you plan to remove restarts? I don't particularly need restarts, just curious.

**Comment by ingesolvoll** [comment]
Well, with the current implementation it will remove the ability to restart. So maybe something in the config that starts the machine? Like `:unmount-when-finished?`.

**Comment by mainej** [comment]
> :unmount-when-finished?

👍  sounds reasonable to me.
